### PR TITLE
aealite agent example and small test

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

aealite agent example and small test

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/go/aealite_agent_example/go.mod [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module aealite_agent_example
+
+go 1.14
+
+replace aealite => ../aealite
+
+require (
+	aealite v0.0.0-00010101000000-000000000000
+	github.com/sirupsen/logrus v1.8.1
```
Comment (DavidMinarsch) [inline libs/go/aealite_agent_example/go.mod:9]: ?

File: libs/go/aealite_agent_example/tests/pexpect_popen.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        while start_time + timeout > time.time() and self.proc.poll() is None:  # type: ignore
+            time.sleep(0.001)
+
+        if self.proc.poll() is None:  # type: ignore
+            self.terminate(force=True)
+            self.wait()
+            self.exitstatus = "Terminated!"  # type: ignore
+
+    @classmethod
+    def aea_cli(cls, args, **kwargs) -> "PexpectWrapper":
```
Comment (DavidMinarsch) [inline libs/go/aealite_agent_example/tests/pexpect_popen.py:75]: Why is this needed here?

File: libs/go/aealite_agent_example/go.mod (hunk 1 above)
Comment (solarw) [reply libs/go/aealite_agent_example/go.mod:9]: requires aealite it self and some logging module to write logs as done in example in aealite README

File: libs/go/aealite_agent_example/tests/pexpect_popen.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+        :param timeout: how many seconds wait process to finish before kill.
+        """
+        if self.proc.poll() is not None:  # type: ignore
+            return
+
+        start_time = time.time()
+
+        while start_time + timeout > time.time() and self.proc.poll() is None:  # type: ignore
+            time.sleep(0.001)
```
Comment (marcofavorito) [inline libs/go/aealite_agent_example/tests/pexpect_popen.py:67]: worth to make `0.001` customizable?

File: libs/go/aealite_agent_example/tests/pexpect_popen.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        :param timeout: timeout in seconds
+
+        :return: None
+        """
+        self.expect(EOF, timeout=timeout)
+
+    def terminate(self, *args, **kwargs) -> None:
+        """Terminate process."""
+        if self.proc.poll() is None:
+            self.kill(signal.SIGTERM)
```
Comment (marcofavorito) [inline libs/go/aealite_agent_example/tests/pexpect_popen.py:128]: Is this guaranteed to terminate the process? What about waiting until SIGTERM has been handled, and then, if the process has not terminated yet, send `SIGKILL`?

**Review by marcofavorito — APPROVED** [review]
LGTM

File: libs/go/aealite_agent_example/tests/pexpect_popen.py (hunk 3 above)
Comment (solarw) [reply libs/go/aealite_agent_example/tests/pexpect_popen.py:67]: does n make sense in this case, sleep done just for periodical check

File: libs/go/aealite_agent_example/tests/pexpect_popen.py (hunk 4 above)
Comment (solarw) [reply libs/go/aealite_agent_example/tests/pexpect_popen.py:128]: switched to sigkill

**Review by DavidMinarsch — APPROVED** [review]
LGTM
